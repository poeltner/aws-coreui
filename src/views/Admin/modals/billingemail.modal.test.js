import React from 'react';
import { shallow, mount } from 'enzyme';

import BillingEmailModal from './billingemail.modal';
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withNamespaces: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: () => "" };
    return Component;
  },
}));

describe('BillingEmailModal', () => {
  // it('should render correctly in "debug" mode', () => {
  //   const component = shallow(<BillingEmailModal debug />);
  //   expect(component).toMatchSnapshot();
  // });

  it('should through an error, if no email in input box', () => {
    let billingEmailModal = React.createRef();
    const wrapper = mount(<BillingEmailModal onRef={ref => (billingEmailModal = ref)} />);
    wrapper.setProps({ modal: true });

    console.log("Compoonent " + JSON.stringify(wrapper));
    const input = wrapper.find('email');
    console.log("input  " + JSON.stringify(input));
    input.simulate('change', { target: { value: 'Hello' } })
    expect(input.get(0).value).to.equal('Hello');
  });
});