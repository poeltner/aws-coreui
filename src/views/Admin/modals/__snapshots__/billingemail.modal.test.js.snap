// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingEmailModal should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadNamespace(BillingEmailModal)
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
